certbot/tests/client_test.py

```diff
@@ -25,15 +27,29 @@ def setUp(self):
         self.config.register_unsafely_without_email = False
         self.config.email = "alias@example.com"
         self.account_storage = account.AccountMemoryStorage()
+        self.config.eab_hmac_key = "hmac-key-for-testing"
```

Comment:
We want these to sometimes not be available, so they should be set in `config` only when explicitly testing that an EAB is constructed.

Reply:
I set them to None in certbot.constants.CLI_DEFAULTS instead now. Seems to have solved the reason I originally set them. 
Still working through some test logic problems in certbot/client.py that was due to a Mock always returning True: "if acme.client.directory.meta.external_account_required".